You place the ace of spades on the bottom of a standard deck of 52 cards. You shuffle the deck (assume in a perfectly random way so that any position afterwards is equally likely) and observe where the ace is now. You then remove all cards under it, which is equivalent to saying the ace can't shuffle back to a lower position. 

Keep repeating this process until the ace is on top of the deck.

What is the expected number of shuffles to achieve this?

In [1]:
import random

n_sims = 10**6
shuffles_required = 0

for _ in range(n_sims):
    ace_position = 52
    while ace_position > 1:
        shuffles_required += 1
        ace_position = random.choice(range(1, ace_position+1))
        
shuffles_required/n_sims

5.521964

A simulation yields ~5.52 shuffles. We can solve this without simulation though.

Let N<sub>i</sub> equal the expected number of shuffles needed when there are `i` cards in the deck.

\begin{equation*}N_1 = 0 \end{equation*}
\begin{equation*}N_2 = 1 + \frac{N_1}{2} + \frac{N_2}{2} \end{equation*}
\begin{equation*}N_3 = 1 + \frac{N_1}{3} + \frac{N_2}{3} + \frac{N_3}{3} \end{equation*}
\begin{equation*}N_4 = 1 + \frac{N_1}{4} + \frac{N_2}{4} + \frac{N_3}{4} + \frac{N_4}{4} \end{equation*}

Let's take a look at `N4`. We know that a shuffle is going to take us to having either 4, 3, 2, or 1 cards left in the deck, all states having equal probability `1/4`. So translating this formula into English we say that the expected number of shuffles needed with 4 cards in the deck is equal to the expected number of shuffles needed with 1, 2, 3, or 4 cards in the deck, times the probility we end up in each of those situations. 

Rearranging the original formula algebraically gets us to:

\begin{equation*}N_4 = \frac{4}{3} + \frac{N_1 + N_2 + N_3}{3} \end{equation*}

If you repeat this step for other values of `i` you can see the general formula is:

\begin{equation*}N_i = \frac{i}{i-1} + \frac{\sum_{j=1}^{i-1} {N_j}}{i-1} \end{equation*}

Calculating the value of a full deck of 52 cards returns...

In [2]:
N = dict()
N[1] = 0

for i in range(2, 53):
    N[i] = i/(i-1) + sum([N[j] for j in range(1, i)])/(i-1)

N[52]

5.518813181466678